tests/e2e/nvidia-ctk_containerd_test.go

```diff
@@ -0,0 +1,850 @@
+		image:             "docker.io/kindest/base:v20250521-31a79fd4",
+		configVersion:     3,
+		pluginPath:        "io.containerd.cri.v1.runtime",
+		hasDefaultImports: true,
+	},
+}
+
+// Integration tests for containerd drop-in config functionality
+var _ = Describe("containerd", Ordered, ContinueOnFailure, Label("container-runtime"), func() {
+	// Run all tests for each containerd version
+	for _, env := range containerdEnvs {
+		env := env // capture loop variable
+
+		Context(fmt.Sprintf("with %s", env.name), Ordered, func() {
+			var (
+				nestedContainerRunner Runner
+				containerName         = fmt.Sprintf("nvctk-e2e-containerd-%s-tests", env.name)
+			)
+
+			// restartContainerdAndWait restarts containerd and waits for it to be ready
+			restartContainerdAndWait := func(runner Runner) error {
+				_, _, err := runner.Run(restartContainerdScript)
+				if err != nil {
+					return fmt.Errorf("failed to restart containerd: %w", err)
+				}
+
+				_, _, err = runner.Run(waitForContainerdScript)
+				if err != nil {
+					return fmt.Errorf("containerd did not become ready after restart")
+				}
+				return nil
+			}
+
+			BeforeAll(func(ctx context.Context) {
+				var err error
+
+				// Create the nested container with the global cache mounted
+				nestedContainerRunner, err = NewNestedContainerRunner(runner, env.image, installCTK, containerName, localCacheDir)
+				Expect(err).ToNot(HaveOccurred())
+
+				// Backup original containerd configuration
+				_, _, err = nestedContainerRunner.Run(`
+			# Backup the original conf.d directory
+			if [ -d /etc/containerd/conf.d ]; then
+				cp -r /etc/containerd/conf.d /tmp/containerd-conf.d.backup
+			fi
+			
+			# Backup the original config.toml
+			if [ -f /etc/containerd/config.toml ]; then
+				cp /etc/containerd/config.toml /tmp/containerd-config.toml.backup
+			fi
+		`)
+				Expect(err).ToNot(HaveOccurred(), "Failed to backup containerd configuration")
+
+				// Ensure containerd is running
+				_, _, err = nestedContainerRunner.Run(`
+					# Start containerd if not running
+					if ! systemctl is-active --quiet containerd; then
+						systemctl start containerd
+						sleep 2
+					fi
+				`)
+				Expect(err).ToNot(HaveOccurred(), "Failed to ensure containerd is running")
+
+				// Version-specific setup
+				if !env.hasDefaultImports {
+					// For containerd 1.7.x, ensure imports directive exists
+					_, _, err = nestedContainerRunner.Run(`set -e
+# Ensure config file exists
+if [ ! -f /etc/containerd/config.toml ]; then
+	mkdir -p /etc/containerd
+	containerd config default > /etc/containerd/config.toml
+fi
+
+# Add imports directive if not present
+if ! grep -q "imports" /etc/containerd/config.toml; then
+	# Create imports line
+	cat > /tmp/imports.line <<EOF
+imports = ["/etc/containerd/conf.d/*.toml"]
+EOF
+	# Prepend to existing config
+	cat /etc/containerd/config.toml >> /tmp/imports.line
+	mv /tmp/imports.line /etc/containerd/config.toml
+fi`)
```

Comment:
Why are we doing this? The `nvidia-ctk runtime configure` command should do this in cases where the imports directive does not exist.